src/main/java/AuthenticationService.java

```diff
@@ -1,3 +1,68 @@
+import java.util.UUID;
+
+/**
+ * Description: This class handles user authentication,
+ * credential verification, and secure system access.
+ *
+ * Intent: The AuthenticationService class contains
+ * the core logic for user authentication.
+ * It provides methods to create users, authenticate users,
+ * change usernames and passwords, and delete user accounts.
+ * It interfaces with the UserRepository
+ * to fetch and update user information.
+ */
 public class AuthenticationService {
-    // TODO implement class
+
+    private UserRepository userRepository;
+
+    public AuthenticationService(UserRepository userRepository) {
+        this.userRepository = userRepository;
+    }
+
+    public User createUser(String username, String password) {
+        UUID id = UUID.randomUUID();
+        return new User(id, username, password);
+    }
+
+    public User changeUsername(UUID id, String newUsername) {
+        if (newUsername == null || newUsername.isEmpty() || newUsername.trim().isEmpty()) {
```

Comment:
```suggestion
        if (newUsername == null || newUsername.trim().isEmpty()) {
```
Duplicated condition